### PR TITLE
Save light curve default image file format

### PR DESCRIPTION
**Pull request by MurrayForbes**
Change the default image file format from bmp to png for the lightcurve plot's File/Export lightcurve/save as image file' code (png uses a lossless image compression but results in smaller file sizes than bmp). Also remember the  image format used last time Tangra was run, and set it as the default image format.

### CODE REVIEW

**Comment by hpavlov** [comment]
Hi Murray,

- Please don't change white space as it makes code review very difficult. The alternative would be to have white space changes only in a separate commit. However I'd rather make the reformatting or the entire code base not on single files. For now I suggest that no white space is changed.
- The naming convention for enums values followed in the code base is Pascal case i.e. start with capital letter. So in your case this could be:

       public enum SaveAsImageFormat
        {
            Png,
            Bmp,
            Jpg
        }

**Comment by MurrayForbes** [comment]
Hristo,

Sorry about the white space, I did it before your email about not doing it on a whole file or project.

Have just changed the enums, and pushed the commit.

Cheers,
Murray.

Sent from Mail for Windows 10

**Comment by hpavlov** [comment]
Because you have already added all the files inside the .vs directory to the commit, excluding the directory in the .gitignore will not be enough. You will also need to remove the files from the commit. 

Another alternative is to do an entirely new commit copying the changes over. Then when you initiate the new commit the .gitignore change will 'kick in' and the files inside .vs will not be added as part of the commit.

**Comment by MurrayForbes** [comment]
Not sure if I’ve done this right, but here goes ....

Sent from Mail for Windows 10

File: Tangra.Model/Config/TangraConfig.cs [hunk 1]
```diff
@@ -1153,7 +1163,10 @@ public Color Target4Color
 
 			public static Font LabelsFont = new Font(FontFamily.GenericMonospace, 9);
 
-			public void Load()
+            public SaveAsImageFormat SaveAsImageType;
```
Comment (hpavlov) [inline Tangra.Model/Config/TangraConfig.cs:1166]: Looks like this field is unused, where `SaveAsImageFileType` is used instead. Do I read this correctly? If so then this line can be deleted.

File: Tangra 3/VideoOperations/LightCurves/frmLightCurve.cs [hunk 2]
```diff
@@ -1551,15 +1551,60 @@ private void miLoad_Click(object sender, EventArgs e)
...
+                SaveLastUsedSaveAsImageFormat(file_ext);
+            }
+        }
+
+        private void SetSaveAsImageFormatToLastUsed()
+        {
+
+            switch (TangraConfig.Settings.Generic.SaveAsImageFileType)
+            {
+                case TangraConfig.SaveAsImageFormat.Bmp:
```
Comment (hpavlov) [inline Tangra 3/VideoOperations/LightCurves/frmLightCurve.cs:1577]: Changing the order of the filter is not really necessary. A better approach would be to use a single filter, but based on the saved setting to auto-suggest the file name with a different extension i.e. to set `saveImageDialog.FileName` before opening the dialog. The name of the file could be the name of the video files with the corresponding extension.

File: Tangra 3/VideoOperations/LightCurves/frmLightCurve.cs [hunk 3]
```diff
@@ -1551,15 +1551,60 @@ private void miLoad_Click(object sender, EventArgs e)
...
+
             Bitmap bmp = AddImageLegend();
 
-			if (saveImageDialog.ShowDialog(this) == DialogResult.OK)
-			{
+            SetSaveAsImageFormatToLastUsed();
+
+            if (saveImageDialog.ShowDialog(this) == DialogResult.OK)
+            {
+                string file_ext = Path.GetExtension(saveImageDialog.FileName);
```
Comment (hpavlov) [inline Tangra 3/VideoOperations/LightCurves/frmLightCurve.cs:1562]: Please use the same naming convention used across the code base. Typically in C# local variables use camel case i.e. `fileExt` in this case. This is also the conversion used in Tangra. Snake case (`file_ext`) is a Python thing as the name suggests :)

Fro the records Tangra also uses a non-standard C++ convention for member variables with an m_CamelCase prefix and s_CamelCase for static ones.

**Comment by MurrayForbes** [comment]
My bad – I’d been trying out a few things (which didn’t work) and forgot to remove everything. 

Cheers,
Murray.

Sent from Mail for Windows 10

**Comment by MurrayForbes** [comment]
Okay, renamed.

I’ve noticed there is a lot of messages for words that violate the naming rules, for example ‘Naming rule violation: these words must begin with upper case characters  frmLightCurve_FormClosed’. 

I think I missed my naming violation in the forest of similar messages.

Cheers,
Murray.

Sent from Mail for Windows 10

**Comment by hpavlov** [comment]
VS or Resharper (if using) may offer their own naming convention. My point was that it is better if the entire code base of an application uses the same naming convention and there are different naming conventions that could be ued. I already know that Tangra's code base does not conform to the VS suggested one or other more commonly used ones. So wanted to point out what the naming convention is (see below). You should stick to that and can ignore messages from VS.
member variables (non-public): prefixed with m_ then PascalCasestatic variables (non-public): prefixed with s_ then PascalCase
public members/properties/functions/enum values: PascalCaselocal variables: camelCaseconstants: ALL_CAPS_SNAKE_CASEvisual controls: camelCase with Hungarian notation prefix for the type of the control, i.e. tbxMyTextBox, cbMyCheckBox, btnMyButtonabbreviations (public): PascalCase (e.g. Id rather than ID, Bmp rather that BMP)


    On Saturday, June 26, 2021, 10:23:55 AM GMT+10, MurrayForbes ***@***.***> wrote:  
 
 

Okay, renamed.

I’ve noticed there is a lot of messages for words that violate the naming rules, for example ‘Naming rule violation: these words must begin with upper case characters frmLightCurve_FormClosed’. 

I think I missed my naming violation in the forest of similar messages.

Cheers,
Murray.

Sent from Mail for Windows 10

**Comment by MurrayForbes** [comment]
Hristo,

Your approach sounds better than what I did. However, I haven’t been able to find the name of the video file (I find lots of m_fileName or similar but none seem to be the right one and/or not accessible in my method).

Cheers,
Murray.

Sent from Mail for Windows 10

**Comment by hpavlov** [comment]
Should be accessible via TangraContext.Current.FileName (I think*)


    On Saturday, June 26, 2021, 8:07:36 PM GMT+10, MurrayForbes ***@***.***> wrote:  
 
 

Hristo,

Your approach sounds better than what I did. However, I haven’t been able to find the name of the video file (I find lots of m_fileName or similar but none seem to be the right one and/or not accessible in my method).

Cheers,
Murray.

Sent from Mail for Windows 10

**Comment by MurrayForbes** [comment]
Hristo,

Yes, that was it. I’ll push this in a few minutes.

I could probably do the same thing (auto-suggesting the file name ) for the CSV and LC files as well?

Cheers,
Murray.

Sent from Mail for Windows 10

**Comment by hpavlov** [comment]
Hi Murray,
> I could probably do the same thing (auto-suggesting the file name ) for the CSV and LC files as well?
This has been already done for those cases I believe, but please check and implement it if not working that way.
Hristo.

    On Sunday, June 27, 2021, 6:03:05 AM GMT+10, MurrayForbes ***@***.***> wrote:  
 
 

Hristo,

Yes, that was it. I’ll push this in a few minutes.

I could probably do the same thing (auto-suggesting the file name ) for the CSV and LC files as well?

Cheers,
Murray.

Sent from Mail for Windows 10

**Comment by MurrayForbes** [comment]
Hristo,

No, it wasn’t already done (if it had been, I’d have used that code as a template and would have seen TangraContext.Current.FileName).

Anyway, I’ve amended the code for the lc, csv and emf file saves to use the video’s filename, and to override that if a lc file has been saved or loaded.

Cheers,
Murray.

Sent from Mail for Windows 10

**Pull request closed without merging**